cdap-watchdog/src/test/java/co/cask/cdap/metrics/collect/AggregatedMetricsCollectionServiceTest.java

```diff
@@ -101,27 +99,18 @@ protected long getPeriodMillis() {
       service.getContext(EMPTY_TAGS).increment(METRIC, 3);
       service.getContext(EMPTY_TAGS).increment(METRIC, 4);
 
-      MetricValues record = published.poll(10, TimeUnit.SECONDS);
-      Assert.assertNotNull(record);
-
-      Assert.assertEquals(((long) Integer.MAX_VALUE) + 9L, getMetricValue(record.getMetrics(), METRIC));
+      verifyCounterMetricsValue(published, ImmutableMap.of(0, ImmutableMap.of(METRIC, 9L + Integer.MAX_VALUE)));
 
       // No publishing for 0 value metrics
       Assert.assertNull(published.poll(3, TimeUnit.SECONDS));
 
-      // Publish a metric and wait for it so that we know there is around 1 second to publish more metrics to test.
```

Comment:
this is the previous poor man's logic to prevent the publishing in the middle of the next few gauge() calls. Turns out that that does not work reliably (e.g. if there is a GC pause). With the new verification logic this is not needed any more.